### PR TITLE
Change Config to structopt enhanced ConfigOpt

### PR DESCRIPTION
**Pull request by notmandatory**
This change sets config values first, if found, from config.ini, these values are overridden by environment variables, and finally overridden by command line arguments. If any required values are missing (eg. descriptor) or an invalid value is given (eg. network) then structopt provides an appropriate error.

The README is also updated to explain how to deploy the server to heroku and configure it using environment vars or the config.ini file.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,19 +9,31 @@ Another Bitcoin payment service, based on [bdk](https://github.com/bitcoindevkit
...
+#### Configured via environment vars
 1. Fork this project 
+2. Press the below button
+3. When prompted enter required configuration values for your node
+
+OR 
+
+#### Configured via .ini file 
+1. Fork the project
 2. Copy `config_default.ini` to `config.ini`
```
Comment (lvaccaro) [inline README.md:34]: nit: could you also update to use `config_example.ini` instead default?

File: config_test.ini [hunk 2]
```diff
@@ -0,0 +1,17 @@
+[BDK]
```
Comment (lvaccaro) [inline config_test.ini:1]: nit: this file is used in the tests. do you suggest to keep it separate from `config_example.json` ?

**Comment by lvaccaro** [comment]
nit: do you suggest to remove `app.json` file in order to keep the `app_ini_example.json`?

File: README.md (hunk 1 above)
Comment (notmandatory) [reply README.md:34]: oops ya this was a mistake, will fix.

File: config_test.ini (hunk 2 above)
Comment (notmandatory) [reply config_test.ini]: I was going to keep it but it is a little confusing having it here. I found a simpler way to do my testing and was able to removed this file.

**Comment by notmandatory** [comment]
> nit: do you suggest to remove `app.json` file in order to keep the `app_ini_example.json`?

My idea is if someone wants to deploy to heroku and config via env variables they can use this `app.json` as it is and don't need to change anything in the project. But if they want to use the .ini config then since they have to copy and edit `config_example.ini` they can also copy `app_ini_example.json` which just removes the required env variables.

**Comment by notmandatory** [comment]
moved to draft while I do some more testing on heroku

**Comment by notmandatory** [comment]
I ran into a hard crash with url `/bitcoin/` but seems to be an existing issue on server.rs line 67. I'll add an issue for it.

**Comment by lvaccaro** [comment]
utack a7d5510827885c82ae64969db0e15f8342244ca0

**Pull request merged**